### ISSUE TITLE
"not explicitly sorted" warning in `join_asof` when `by` specified

### ISSUE THREAD

**Issue by stinodego** [pola-rs/polars#8981]
### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://crates.io/crates/polars) of Polars.

### Issue description

I cannot seem to disable these warnings, not even `set_sorted` works.

My guess is that the initial join on `by` unsets the sorted flag.

### Reproducible example

```python
import polars as pl

lf1 = pl.LazyFrame(
    schema={
        "id": pl.Int16,
        "timestamp": pl.Datetime,
    },
).sort("timestamp")


lf2 = pl.LazyFrame(
    schema={
        "id": pl.Int16,
        "timestamp": pl.Datetime,
        "amount": pl.Int32,
    },
).sort("timestamp")

lf2.join_asof(lf1, by=["id"], on="timestamp").collect()
# argument in operation 'asof_join' is not explicitly sorted
```

### Expected behavior

No warnings.

### Installed versions

<details>
--------Version info---------
Polars:      0.17.14
Index type:  UInt32
Platform:    Linux-5.10.102.1-microsoft-standard-WSL2-x86_64-with-glibc2.35
Python:      3.11.2 (main, Mar 31 2023, 16:17:48) [GCC 11.3.0]

----Optional dependencies----
numpy:       1.24.3
pandas:      <not installed>
pyarrow:     12.0.0
connectorx:  <not installed>
deltalake:   0.8.1
fsspec:      2023.5.0
matplotlib:  <not installed>
xlsx2csv:    <not installed>
xlsxwriter:  <not installed>
</details>

**Comment by CloseChoice** [comment]
Can't confirm this on the latest main.

**Comment by stinodego** [comment]
You're right - must have already been fixed after the `0.17.14` release. Thanks!
